builders/get-started/eth-compare/consensus-finality.md

```diff
@@ -0,0 +1,78 @@
+
+This guide will outline some of these main differences around consensus and finality, and what to expect when using Moonbeam for the first time.
+
+## Ethereum Consensus and Finality
+
+As stated before, Ethereum is currently using a PoW consensus protocol and the longest chain rule, where finality is probabilistic. 
+
+Probabilistic finality means that the probability that a block (and all its transactions) will not be reverted increases as more blocks are built on top of it. Therefore, the higher the number of blocks you wait, the higher the certainty that a transaction will not be re-organized, and consequently reverted. As suggested by [this blog on finality](https://blog.ethereum.org/2016/05/09/on-settlement-finality/) by Vitalik, _"you can wait 13 confirmations for a one-in-a-million chance of the attacker succeeding."_
+
+## Moonbeam Consensus and Finality
+
+Moonbeam uses a block production mechanism based on the [Polkadot NPoS consensus](https://wiki.polkadot.network/docs/en/learn-consensus). In such a mechanism, there are collators and validators. [Collators](https://wiki.polkadot.network/docs/en/learn-collator) maintain parachains (in this case, Moonbeam) by collecting transactions from users and producing state transition proofs for the Relay Chain [validators](https://wiki.polkadot.network/docs/en/learn-validator). The collators set (nodes that produce blocks) are selected based on the [stake they have in the network](/learn/features/consensus/). 
+
+For finality, Polkadot/Kusama rely on [GRANDPA](https://wiki.polkadot.network/docs/learn-consensus#finality-gadget-grandpa). GRANDPA provides deterministic finality for any given transaction (block). In other terms, when a block/transaction is marked as final, it can't be reverted except via on-chain governance or forking.
+
+## Main Differences
+
+In terms of consensus, Moonbeam is based on Nominated Proof-of-Stake, while Ethereum relies on Proof-of-Work, which are very different. Consequently, Proof of Work concepts, such as  `difficulty`, `uncles`, `hashrate`, generally don’t have meaning within Moonbeam.
+
+For APIs that return values related to Ethereum’s Proof of Work, default values are returned. Existing Ethereum contracts that rely on Proof of Work internals (e.g., mining pool contracts) will almost certainly not work as expected on Moonbeam.
+
+In terms of finality, on Moonbeam, you can check when a transaction is finalized, meaning that it can't be reverted. The strategy is fairly simple: 
+ 1. You ask the network the hash of the latest finalized block
+ 2. You retrieve the block number using the hash
+ 3. You compare it with the block number of your transaction. If your transaction was included in a previous block, it is finalized
```

Comment:
This is a good quick-n-dirty technique that will work most of the time, but it does not guarantee finality. This only checks that your tx is in a block with a lower number than the latest finalized. It does not guarantee that your tx in in the same chain. For example imagine if your tx goes in #10, but there is a fork at #9. Later a block #11 is finalized, but it is not built on your block. Your tx got orphaned.

Sadly, I don't really know the RPCs well enough to know how you can get this info more reliably.

Reply:
After https://github.com/paritytech/frontier/pull/474 - we only return tx receipts for tx included in canonical blocks. Is it still possible to get re-orged? I don't think so right?

Reply:
Until we have a dedicated RPC method, the methodology above is "valid" because we check for block finality first, then get the tx receipt and verify that the block number is smaller than the finalized block. For redundancy, I'll include a `eth_getBlockByNumber` after to verify that the tx hash is included